main.tf

```diff
@@ -290,6 +291,39 @@ data "aws_iam_policy_document" "bucket_policy" {
       }
     }
   }
+
+  dynamic "statement" {
+    for_each = length(var.replication_source_roles) > 0 ? [1] : []
+
+    content {
+      sid = "CrossAccountReplicationObjects"
+      actions = [
+        "s3:ReplicateObject",
+        "s3:ReplicateDelete",
```

Comment:
I would also consider making `s3:ReplicateDelete` conditional based on `var.replication_delete_enabled`.

For example if this is a centralized bucket for audit logs, it's probably best to make `s3:ReplicateDelete` conditional.

Reply:
You mean add a new variable? Since there might be multiple entries in `var.replication_source_roles`, I think we would need to adjust the variable to be like this:

```terraform
variable "replication_source_roles" {
  type        = list(object({arn = string, replicate_delete_permission = bool})
  default     = []
}
```

IMO, this is too complex. If users want to add permissions for a locked down role like this, they can add it by hand. It's not worth making the API for this module more complex.

If you think it's still a good idea, I will implement though. Let me know.

Reply:
I think I agree with you that we should keep it as simple as possible. I suggested it because I've seen a real world case where this is needed, but as you implied, the user can build out the policy manually and pass it via `var.policy`.